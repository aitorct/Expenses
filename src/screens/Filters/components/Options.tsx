import React, {useCallback} from 'react';
import {FlexStyle, StyleProp, StyleSheet, View} from 'react-native';
import {FilterValue} from '@contexts/FiltersContext/types';
import TouchableTag from '@components/TouchableTag';
import fontSizes from '@theme/fontSizes';
import sizes from '@theme/sizes';

interface Props {
  options: string[];
  activeOption: FilterValue;
  onPress: (option: string) => void;
  itemsPerRow?: number;
}

const Options = ({options, activeOption, onPress, itemsPerRow = 3}: Props) => {
  const getComputedStyles = useCallback(
    (index: number): StyleProp<FlexStyle> => ({
      marginRight: (index + 1) % itemsPerRow === 0 ? undefined : sizes.s8,
      marginBottom:
        Math.floor(index / itemsPerRow + 1) <
        Math.ceil(options.length / itemsPerRow)
          ? sizes.s8
          : undefined,
    }),
    [itemsPerRow, options.length],
  );

  return (
    <View style={[styles.container]}>
      {options.map((option, index) => (
        <TouchableTag
          key={index}
          onPress={() => onPress(option)}
          text={option}
          active={option === activeOption}
          textStyle={styles.tagText}
          buttonStyle={getComputedStyles(index)}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    flexWrap: 'wrap',
  },
  tagText: {
    fontSize: fontSizes.largeText,
  },
});

export default Options;
