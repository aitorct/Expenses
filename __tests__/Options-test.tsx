import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Options from '@screens/Filters/components/Options';

const mockedFunction = jest.fn(() => {});

const options = ['1', '2', '3'];

it('FilterHeader with no active option renders correctly', () => {
  const tree = renderer
    .create(
      <Options
        options={options}
        activeOption={null}
        onPress={mockedFunction}
        itemsPerRow={3}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Single row FilterHeader renders correctly', () => {
  const tree = renderer
    .create(
      <Options
        options={options}
        activeOption={options[0]}
        onPress={mockedFunction}
        itemsPerRow={3}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Multiple row FilterHeader renders correctly', () => {
  const tree = renderer
    .create(
      <Options
        options={options}
        activeOption={options[0]}
        onPress={mockedFunction}
        itemsPerRow={2}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
