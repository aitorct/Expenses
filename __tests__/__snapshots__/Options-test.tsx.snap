// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterHeader with no active option renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": undefined,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Multiple row FilterHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": 8,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "borderColor": "#74a2e2",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": 8,
        "marginRight": undefined,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Single row FilterHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "borderColor": "#74a2e2",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": undefined,
        "marginRight": undefined,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7EFFA",
            "borderRadius": 36,
            "flexDirection": "row",
            "padding": 4,
            "paddingHorizontal": 8,
          },
          Array [
            Object {
              "borderWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            Object {
              "backgroundColor": "#7D82851A",
              "borderColor": "#7D82851A",
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
</View>
`;
